src/program/services/scrapers/scraper_base.py

```
@@ -0,0 +1,62 @@
+        return identifier, scrape_type, imdb_id
+
+    @staticmethod
+    def get_infohash_from_torrent_url(url: str) -> str:
+        """Get the infohash from a torrent URL"""
+        session = SmartSession()
+        with session.get(url, stream=True) as r:
+            r.raise_for_status()
+            torrent_data = r.content
+            torrent_dict = decode(torrent_data)
+            info = torrent_dict[b"info"]
+            infohash = hashlib.sha1(encode(info)).hexdigest()
```

Ruff (0.13.2)

Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)